### ISSUE TITLE
forecast_pred.py

### ISSUE THREAD

**Issue by jai2033shankar** [jai2033shankar/inventory_forecasting#11]
Here is the feedback on the code:

**About the code:**
The code appears to be a Python script that uses various libraries, including NumPy, Pandas, SciPy, and Statsmodels, to analyze and forecast a time series dataset. The script performs various tasks, including data loading, preprocessing, model fitting, and prediction. The goal of the script is to develop an ARIMA model to forecast a time series dataset.

**Status:**
I would rate the overall quality of the code as good. The code is well-structured and easy to follow. However, there are some areas where improvements can be made.

**Issues:**

1. **Code readability:** The code could benefit from more descriptive variable names and comments to explain the purpose of certain sections.
2. **Code organization:** The code is a bit long and could be broken down into smaller functions to improve organization and reusability.
3. **Data handling:** The code assumes that the data is already cleaned and preprocessed, but it would be better to include data cleaning and preprocessing steps in the code.
4. **Model evaluation:** The code only evaluates the model using the mean squared error (MSE) metric, but it would be better to include other metrics, such as mean absolute error (MAE) and mean absolute percentage error (MAPE), to get a more comprehensive understanding of the model's performance.
5. **Code duplication:** There are some duplicate code sections, which could be refactored to reduce code duplication and improve maintainability.

**Improved code:**
Here is an improved version of the code that addresses some of the issues mentioned above:
```python
import pandas as pd
import numpy as np
from statsmodels.tsa.arima.model import ARIMA
from sklearn.metrics import mean_squared_error, mean_absolute_error, mean_absolute_percentage_error

# Load data
dataset = pd.read_csv('dataset.csv', header=None, index_col=0, parse_dates=True, squeeze=True)

# Preprocess data
X = dataset.values.astype('float32')
train_size = int(len(X) * 0.50)
train, test = X[0:train_size], X[train_size:]

# Define a function to evaluate the model
def evaluate_model(model, X, y):
    yhat = model.forecast()[0]
    error = y - yhat
    mse = mean_squared_error(y, yhat)
    mae = mean_absolute_error(y, yhat)
    mape = mean_absolute_percentage_error(y, yhat)
    return mse, mae, mape

# Fit the ARIMA model
model = ARIMA(train, order=(0,0,1))
model_fit = model.fit()

# Make predictions
predictions = []
for i in range(len(test)):
    yhat = model_fit.forecast()[0]
    predictions.append(yhat)

# Evaluate the model
mse, mae, mape = evaluate_model(model, test, predictions)
print('MSE: %.3f' % mse)
print('MAE: %.3f' % mae)
print('MAPE: %.3f' % mape)

# Plot the results
pyplot.plot(test)
pyplot.plot(predictions, color='red')
pyplot.show()
```
This improved code includes more descriptive variable names, breaks down the code into smaller functions, and includes additional metrics for model evaluation.
